tests/fv/fv_test.go

```diff
@@ -85,52 +86,112 @@ var _ = Describe("PolicyController", func() {
+				return err
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(HaveOccurred())
+		})
+		It("should update existing profiles in etcd to match namespaces in k8s", func() {
+			profile, err := calicoClient.Profiles().Update(&api.Profile{Metadata: api.ProfileMetadata{Name: profName}})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(profile.Metadata.Labels).To(BeNil())
+			Eventually(func() map[string]string {
+				prof, _ := calicoClient.Profiles().Get(api.ProfileMetadata{Name: profName})
+				return prof.Metadata.Labels
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(BeEmpty())
+		})
+	})
+
+	Context("policies", func() {
+		var policyName string
+		var genPolicyName string
+
+		BeforeEach(func() {
+			// Create a Kubernetes NetworkPolicy.
+			policyName = "jelly"
+			genPolicyName = "knp.default.default." + policyName
+			var np *extensions.NetworkPolicy
+			np = &extensions.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      policyName,
+					Namespace: "default",
+				},
+				Spec: extensions.NetworkPolicySpec{
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"fools": "gold",
+						},
+					},
+				},
+			}
+			err := k8sClient.Extensions().RESTClient().
+				Post().
+				Resource("networkpolicies").
+				Namespace("default").
+				Body(np).
+				Do().Error()
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for it to appear in Calico's etcd.
+			Eventually(func() *api.Policy {
+				policy, _ := calicoClient.Policies().Get(api.PolicyMetadata{Name: genPolicyName})
+				return policy
+			}).ShouldNot(BeNil())
+		})
+
+		It("should write new policies in etcd to match policies in k8s", func() {
+			// Delete the Policy.
+			err := calicoClient.Policies().Delete(api.PolicyMetadata{Name: genPolicyName})
 			Expect(err).ShouldNot(HaveOccurred())
 
-			Start(apiserver)
+			// Wait for the policy-controller to write it back.
 			Eventually(func() error {
-				_, err := calicoClient.Profiles().Get(api.ProfileMetadata{Name: "k8s_ns.peanutbutter"})
+				_, err := calicoClient.Policies().Get(api.PolicyMetadata{Name: genPolicyName})
 				return err
 			}, time.Second*15, 500*time.Millisecond).ShouldNot(HaveOccurred())
 		})
+
+		It("should update existing policies in etcd to match policies in k8s", func() {
+			// Change the selector of the policy in etcd.
+			_, err := calicoClient.Policies().Update(&api.Policy{
+				Metadata: api.PolicyMetadata{Name: genPolicyName},
+				Spec: api.PolicySpec{
+					Selector: "calico/k8s_ns == 'default' && ping == 'pong'",
+				},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Wait for the policy-controller to set it back to its original value.
+			Eventually(func() string {
+				p, _ := calicoClient.Policies().Get(api.PolicyMetadata{Name: genPolicyName})
+				return p.Spec.Selector
+			}, time.Second*15, 500*time.Millisecond).ShouldNot(Equal("calico/k8s_ns == 'default' && ping == 'pong'"))
```

Comment:
A stronger assertion is is to use a `Should(Equal(<original selector>))` - any reason we can't do that?